experimental/oak_baremetal_kernel/src/mm/frame_allocator.rs

```diff
@@ -0,0 +1,142 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use super::bitmap_frame_allocator::BitmapAllocator;
+use x86_64::structures::paging::{
+    frame::PhysFrameRangeInclusive, FrameAllocator, FrameDeallocator, PageSize, PhysFrame,
+    Size2MiB, Size4KiB,
+};
+
+/// Allocator to track physical memory frames.
+///
+/// The basic unit we track is a 2 MiB frame. If necessary, we will take one 2 MiB frame and break
+/// it into smaller, 4 KiB frames to satisfy demand.
+///
+/// The parameter N needs to be set to the number of u64-s required to track the physical memory in
+/// a bitmap. For example, if we want to track 128 GiB of memory, that means 128G/2M = 65K frames,
```

Comment:
(nit, not entirely sure)

```suggestion
/// a bitmap. For example, if we want to track 128 GiB of memory, that means 128Gi/2Mi = 65K frames,
```